_posts/2020-10-16-kaitai-struct-v0.9-released.md

```diff
@@ -13,41 +13,43 @@ categories: news
 After a lot of time and effort, Kaitai project is happy to announce
 release of new major version of Kaitai Struct, declarative markup
 language to describe various binary data structures — binary file
-formats, network stream packets, etc.
+formats, network packets, etc.
 
 The basic idea of Kaitai Struct is that a particular format can be
-described using Kaitai Struct language (in a `.ksy` file), which then
-can be compiled using `kaitai-struct-compiler` into source files in
-one of the supported programming languages. These modules will include
-a generated code for a parser that can read described data structure
-from a file / stream and provide access to its contents in a nice,
+described using Kaitai Struct language (in a `.ksy` file, that stands
+for Kaitai Struct YAML), which then can be compiled using
+`kaitai-struct-compiler` into source files in one of the supported
+programming languages. These modules will include a generated
+code for a parser that can read described data structure from a
+file / stream and provide access to its contents in a nice,
 easy-to-comprehend API.
 
 With the previous 0.8 release, Kaitai project celebrated 1000 stars on GitHub,
 and until 0.9 version, it has collected more than [2000 stars](https://github.com/kaitai-io/kaitai_struct/stargazers).
 Thank you all for your support!
 
 This version introduces C++11 target (which uses smart pointers),
-several handy features (like `valid`ations and little-endian bit integers),
-fixes a lot of bugs and includes quite a few infrastructure improvements.
+several handy features (like `valid`ations and little-endian "bit-sized
+types"), fixes a lot of bugs and includes quite a few infrastructure
+improvements.
 
 ## Release highlights
 
 * New targets support:
-  * Python with [Construct](https://construct.readthedocs.io) library
-  * HTML - intended for documentation, preliminary support
-  * Nim - entry-level support (51% tests pass score)
-* New KSY language features:
+  * Python with [Construct](https://construct.readthedocs.io) library ([#377](https://github.com/kaitai-io/kaitai_struct/issues/377))
+  * HTML - intended for documentation, preliminary support ([ec21bdf62bcc9211e514dd8413ed5f6714a8e6bc](https://github.com/kaitai-io/kaitai_struct_compiler/commit/ec21bdf62bcc9211e514dd8413ed5f6714a8e6bc))
+  * Nim - entry-level support (51% tests pass score) ([#619](https://github.com/kaitai-io/kaitai_struct/issues/619))
+* New KS language features:
```

Comment:
No, I don't think that renaming _KSY language_ to _KS language_ brings any gain here, it IMHO merely opens a room for ambiguities. The language is simply called **KSY**, and it IMHO doesn't matter that the reader doesn't know what does the abbreviation comes from, it's just a name. The same logic applies to names _YAML_, _PHP_, _JSON_, etc. - it's not necessary to know what each letter mean.

OTOH, _Kaitai Struct language_ used above is also OK, because the meaning is "the DSL that comes with the KS project". Using _Kaitai Struct YAML language_ would just unnecessarily raise question "WTF is YAML?" in the reader, and it's best to just avoid it. It's not necessary to understand the point of the project.

Reply:
I mean there is no KSY language. The language is KS and it is defined not in terms of YAML, it can be serialized into JSON (`ksj` extension), "PSON" (python syntax for literals (can be securely parsed using literal_eval), there is no official name, so I have named it so, `ksp` extension), BSON(`ksbb` extension), CBOR (`ksbc` extension), MSGPack (`ksbm` extension) (I have created an own metalanguage for describing grammars built upon AST of stuff serializeable into JSON, the idea I have borrowed from KS, and used an own transformerz python library for composable stuff, so in a little of code I can easily support such variations, and IMHO it may make sense to define that impls of KS can do the same) or even a custom binary format (`ksbn` extension, `n` stands for `native`). YAML is only a one of a serialization schemes. So IMHO we should not call the language `KSY`, just `Kaitai Struct` abbreviated as `KS`.

Reply:
All right, let it be.